### PR TITLE
Tic tac toe api

### PR DESCRIPTION
**Pull request by marcalmon**
## Summary

- Completed the card API II - Services

## Screenshots

Screenshot from 2019-07-16 10-58-55

## Trello Card

https://trello.com/c/m97la2X3/10-api-ii-services

### CODE REVIEW

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import React, { Component } from 'react';
+
+import MatchesService from '../../../../../services/MatchesService';
+
+import styles from './styles.module.scss';
+
+const Spinner = require('react-spinkit');
+
+const getWinnerClass = isWinner => (isWinner ? styles.winner : '');
```
Comment (mariadelacamara) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js]: Please extract this function in utils file.

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  componentDidMount() {
+    const match = MatchesService.getMatches();
+    match.then(result => {
+      this.setState({ data: result.data, loading: false });
+    });
+  }
+
+  render() {
+    return (
+      <div>
```
Comment (mariadelacamara) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js]: If `<div>` doesn't have a class or style but you do really need it, I would suggest using `<Fragment>`
It has some advantages, such as:

> It’s a tiny bit faster and has less memory usage (no need to create an extra DOM node). This only has a real benefit on very large and/or deep trees, but application performance often suffers from death by a thousand cuts. This is one cut less.
> Some CSS mechanisms like Flexbox and CSS Grid have a special parent-child relationship, and adding divs in the middle makes it hard to keep the desired layout while extracting logical components

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        <h1>Match History</h1>
+        {this.state.loading ? (
```
Comment (mariadelacamara) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js]: destruct state! :)

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import React, { Component, Fragment } from 'react';
+
+import MatchesService from '../../../../../services/MatchesService';
+import { getWinnerClass } from '../../utils';
+
+const Spinner = require('react-spinkit');
```
Comment (nicolobruno) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js]: This form is very old. In ES6 we do `import Spinner from 'react-spinkit';`

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+class Matches extends Component {
+  state = {
+    data: [],
+    loading: true
+  };
+
+  componentDidMount() {
+    const match = MatchesService.getMatches();
```
Comment (nicolobruno) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js]: Mmm `getMatches()` is plural and you set result in the const `match` that is in singular  👀

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/styles.module.scss [hunk 6]
```diff
@@ -0,0 +1,5 @@
+@import 'src/scss/colors.scss';
+
+.winner {
+  color: $blue;
+}
```
Comment (mariadelacamara) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/styles.module.scss]: add empty line

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    return (
+      <Fragment>
+        <h1>Match History</h1>
+        {loading ? (
+          <Spinner name="three-bounce" color="#00ADEE" />
+        ) : (
+          <ol>
+            {data.map(item => (
+              <li key={item.id}>
+                <span className={getWinnerClass(item.winner === 'player_one')}>{item.player_one}</span> -{' '}
```
Comment (nicolobruno) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js]: Extract `player_one` in constants

File: react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+                <span className={getWinnerClass(item.winner === 'player_two')}>{item.player_two}</span>
```
Comment (nicolobruno) [inline react/tic-tac-toe/src/app/screens/Game/components/Matches/index.js]: Extract `player_two` in constants

**Pull request merged**